### PR TITLE
[MDEP-960] Repair silent logging

### PR DESCRIPTION
**Pull request by elharo**
This test could never fail

### CODE REVIEW

**Review by michael-o — COMMENTED** [review]
I honestly don't udnerstand why this is needed at all. I mean we have `-q` and `-X`...

**Comment by michael-o** [comment]
> Yes, I'm not sure why this was done like this in the first place. This has been around since 2.0 so who knows? I am confident the way it was done is buggy, so this tries to repair as much of it as I easily could. I noticed because the tests mixed up two different silent log classes. Someone wasn't following test first practices when they wrote this.
> 
> If we're comfortable with some incompatibility, then we could mke the silent parameter and associated methods noops.

Maybe we should deprecate for removal in 4?

**Comment by elharo** [comment]
OK, deprecated

**Review by elharo — COMMENTED** [review]
Oh dman it, that went to the wrong branch. Why, why don't we protect our branches?

**Comment by elharo** [comment]
Ping

File: src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java [hunk 1]
```diff
@@ -86,8 +87,10 @@ public abstract class AbstractDependencyMojo extends AbstractMojo {
     /**
      * If the plugin should be silent.
      *
+     * @deprecated to be removed in 4.0; use -Q command line option instead
```
Comment (michael-o) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java:90]: `-q`, not `-Q`

File: src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java [hunk 2]
```diff
@@ -170,18 +173,24 @@ public void setSkip(boolean skip) {
...
+     * @deprecated to be removed in 4.0
      */
+    @Deprecated
     protected final boolean isSilent() {
         return silent;
     }
 
     /**
      * @param silent {@link #silent}
+     * @deprecated to be removed in 4.0; no API replacement, use -Q command line option instead
```
Comment (michael-o) [inline src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java:185]: same here

**Review by michael-o — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java (hunk 1 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java]: done

File: src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java (hunk 2 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/dependency/AbstractDependencyMojo.java]: done

**Review by michael-o — APPROVED** [review]

**Pull request merged**